From bc217a72127dd8243b79c3a5720046e577cb4c4f Mon Sep 17 00:00:00 2001
From: Arpit Agarwal <arp@apache.org>
Date: Sun, 14 Jun 2015 16:24:30 -0700
Subject: [PATCH 1181/2848] HDFS-8595. TestCommitBlockSynchronization fails in
 branch-2.7. (Patch applies to all branches).
 (Contributed by Arpit Agarwal)

(cherry picked from commit 1145d09e0a9fa5d2b21266c4a2c14b8359e623ec)

Change-Id: I9b34dbd3eae7a87aebc0e5011337b3637323be11
---
 .../namenode/TestCommitBlockSynchronization.java   |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestCommitBlockSynchronization.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestCommitBlockSynchronization.java
index 365520d..a34d5e4 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestCommitBlockSynchronization.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestCommitBlockSynchronization.java
@@ -71,6 +71,7 @@ private FSNamesystem makeNameSystemSpy(Block block, INodeFile file)
     blockInfo.initializeBlockRecovery(genStamp);
     doReturn(true).when(file).removeLastBlock(any(Block.class));
     doReturn(true).when(file).isUnderConstruction();
+    doReturn(new BlockInfo[1]).when(file).getBlocks();
 
     doReturn(blockInfo).when(namesystemSpy).getStoredBlock(any(Block.class));
     doReturn("").when(namesystemSpy).closeFileCommitBlocks(
-- 
1.7.9.5

